ts_src/wallet.ts

```diff
@@ -0,0 +1,55 @@
+import * as bip32 from 'bip32';
+import * as tapyrus from 'tapyrusjs-lib';
+import { DataStore } from './data_store';
+import { KeyStore } from './key_store';
+
+export default interface Wallet {
+  keyStore: KeyStore;
+  dataStore: DataStore;
+
+  // Import private key into this wallet.
+  import(key: string): void;
+
+  // Import WIF format private key into this wallet.
+  importWif(wif: string): void;
+
+  // Synchonize utxos with index server.
+  // update(): Promise<void>;
+
+  // Broadcast Transaction
+  // broadcast(tx: tapyrus.Transaction): Promise<string>;
+
+  // Return amount for the specified colorId
+  // balance(colorId?: string): number;
+}
+
+export class BaseWallet implements Wallet {
+  keyStore: KeyStore;
+  dataStore: DataStore;
+  constructor(keyStore: KeyStore, dataStore: DataStore) {
+    this.keyStore = keyStore;
+    this.dataStore = dataStore;
+  }
+
+  import(key: string): void {
+    const restored = bip32.fromBase58(key);
+    this.keyStore.add(restored.privateKey!);
```

Comment:
Isn't it more convenient to retain the extended private key data so that the key can be derived in the future?

Reply:
Fixed 47f292e

Reply:
Save private keys as WIF format, and extended private keys as Base58 (xpriv) format.

95dc360